### ISSUE TITLE
Unexpected token `..` when using spread operator in function arguments

### ISSUE THREAD

**Issue by jackfranklin** [eslint/eslint#2346]
The following code:

```js
function mergeAndConcat(...objects) {
   // stuff
}
```

Is throwing `error Unexpected token ..` on line 3, which is the function definition. I've got the `spread` option set to `true` in the `ecmaFeatures` section of my config, so I wouldn't expect the above to cause an error.

Thanks in advance :)

Jack

**Comment by lo1tuma** [comment]
That’s not a `spread` operator, it’s a `rest` parameter. You need to enable the `restParams` flag.

Somehow `restParams` is missing in the [documentation](http://eslint.org/docs/user-guide/configuring), but it is mentioned in the documentation of [espree](https://github.com/eslint/espree#usage).

**Comment by jackfranklin** [comment]
Aha, thank you! I'll turn that on. I'm quite happy to get a PR in to add it to the documentation if I get a chance.

**Comment by andy-hanson** [comment]
I found this confusing because when `spread` is on, `(arg, ...args) => { }` works, but `(...args) => { }` does not!

**Comment by michaelficarra** [comment]
That shouldn't be allowed. Can you open an issue on espree?

**Comment by andy-hanson** [comment]
eslint/espree#130

**Comment by nzakas** [comment]
This has been fixed in Espree and a new release pushed. If you reinstall ESLint, this should be fixed.

**Comment by datatypevoid** [comment]
+1 @lo1tuma

**Comment by nzakas** [comment]
I believe all concerns have been addressed here, so closing.
